lib/bolt/config.rb

```diff
@@ -1,37 +1,143 @@
 require 'logger'
+require 'yaml'
 
 module Bolt
   Config = Struct.new(
     :concurrency,
     :format,
-    :insecure,
     :log_destination,
     :log_level,
-    :password,
-    :run_as,
-    :sudo,
-    :sudo_password,
+    :modulepath,
     :transport,
-    :key,
-    :tty,
-    :user
+    :transports
   ) do
+
     DEFAULTS = {
       concurrency: 100,
-      tty: false,
-      insecure: false,
       transport: 'ssh',
+      format: 'human',
       log_level: Logger::WARN,
       log_destination: STDERR
     }.freeze
 
+    TRANSPORT_OPTIONS = %i[insecure password run_as sudo sudo_password key tty user].freeze
+
+    TRANSPORT_DEFAULTS = {
+      insecure: false,
+      tty: false
+    }.freeze
+
+    TRANSPORTS = %i[ssh winrm pcp].freeze
+
     def initialize(**kwargs)
       super()
       DEFAULTS.merge(kwargs).each { |k, v| self[k] = v }
+
+      self[:transports] ||= {}
+      TRANSPORTS.each do |transport|
+        unless self[:transports][transport]
+          self[:transports][transport] = {}
+        end
+        TRANSPORT_DEFAULTS.each do |k, v|
+          unless self[:transports][transport][k]
+            self[:transports][transport][k] = v
+          end
+        end
+      end
+    end
+
+    def default_path
+      path = ['.puppetlabs', 'bolt.yml']
+      root_path = '~'
+      if ENV['OS'] == 'Windows_NT'
+        unless root_path == ENV['USERPROFILE']
```

Comment:
I'm not sure I understand what this code is doing.

Reply:
Attempting to discover the '~' equivalent on windows to load the config file. I have not tested it yet.
 It's copied from https://github.com/puppetlabs/leatherman/blob/master/file_util/src/file.cc#L115

Reply:
ruby should expand `~` on Windows based on those environment variables, e.g. `C:\Users\Administrator`

Reply:
ah that's much better. i don't know why I thought it didn't